dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -239,6 +252,67 @@ def test_correct_duplicate_row_count_test(self):
         self.assertEqual(2999, self.trained_schema.total_samples)
         self.assertEqual(0.0, self.trained_schema._get_duplicate_row_count())
 
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler')
+    def test_correlation(self, *mock):
+        # data with one column, correlation not updated
+        data = pd.DataFrame([1.0, None, 1.0, None, 5.0])
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.correlation_matrix)
+
+        # data with two columns, but only one is numerical,
+        # then correlation is not updated
+        data = pd.DataFrame([
+            ['test1', 1.0],
+            ['test2', None],
+            ['test1', 1.0],
+            [None, None]])
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.correlation_matrix)
+
+        # data with multiple numerical columns
+        data = pd.DataFrame({'a': [3, 2, 1, 7, 5, 9, 4, 10, 7, 2],
+                             'b': [10, 11, 1, 4, 2, 5, 6, 3, 9, 8],
+                             'c': [1, 5, 3, 5, 7, 2, 6, 8, 1, 2]})
+        profiler = dp.StructuredProfiler(data)
+        expected_corr_mat = data.corr()
+        pd.util.testing.assert_frame_equal(expected_corr_mat,
+                                           profiler.correlation_matrix)
+
+        # data with multiple numerical columns, with nan values
+        data = pd.DataFrame({'a': [np.nan, np.nan, 1, 7, 5, 9, 4, 10, 7, 2],
+                             'b': [10, 11, np.nan, 4, 2, 5, 6, 3, 9, 8],
+                             'c': [1, 5, 3, 5, 7, 2, 6, 8, np.nan, np.nan]})
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.correlation_matrix)
+
+        # data with only one numerical columns without nan values
+        data = pd.DataFrame({'a': [np.nan, np.nan, 1, 7, 5, 9, 4, 10, 7, 2],
+                             'b': [10, 11, np.nan, 4, 2, 5, 6, 3, 9, 8],
+                             'c': [1, 5, 3, 5, 7, 2, 6, 8, np.nan, np.nan]})
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.correlation_matrix)
+
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler',
+                spec=StructuredDataLabeler)
+    def test_merge_correlation(self, *mock):
+        data = pd.DataFrame({'a': [3, 2, 1, 7, 5, 9, 4, 10, 7, 2],
+                             'b': [10, 11, 1, 4, 2, 5, 6, 3, 9, 8],
+                             'c': [1, 5, 3, 5, 7, 2, 6, 8, 1, 2]})
+        data1 = data[:int(len(data) / 2)]
+        data2 = data[int(len(data) / 2):]
+
+        profile1 = dp.StructuredProfiler(data1)
+        profile2 = dp.StructuredProfiler(data2)
+        merged_profile = profile1 + profile2
```

Comment:
this seems dangerous to test with the exact same data. I would instead suggest having data1 and data2 be different to ensure it is correct.

Reply:
Probably simplify the data significantly too so that you can manually calculate the expected correlations as oppose to getting the result and just putting that as the test. (e.g. if code was actually incorrect and the result was just copied the test would pass but be wrong).

Reply:
data1 and data2 are different
```
data1 = data[:int(len(data) / 2)]
data2 = data[int(len(data) / 2):]
```